src/python/gudhi/cubical_complex.pyx

```diff
@@ -259,6 +260,51 @@ cdef class CubicalComplex:
 
         return output
 
+    def vertices_of_persistence_pairs(self):
+        """This function returns a list of pairs of vertices corresponding to
+        the persistence birth and death cells of the filtration. The cells are represented by
+        their indices in the input list of vertices (and not their indices in the
+        internal datastructure that includes non-minimal cells). Note that when two adjacent
+        vertices have the same filtration value, we arbitrarily return one of the two
+        when calling the function on one of their common faces.
+
+        :returns: The vertices of the positive and negative cells,
+            together with the corresponding homological dimension, in two lists of numpy arrays of integers.
+            The first list contains the regular persistence pairs, grouped by dimension.
+            It contains numpy arrays of shape [number_of_persistence_points, 2].
+            The indices of the arrays in the list correspond to the homological dimensions, and the
+            integers of each row in each array correspond to: (index of positive vertex,
+            index of negative vertex).
+            The second list contains the essential features, grouped by dimension.
+            It contains numpy arrays of shape [number_of_persistence_points, 1].
+            The indices of the arrays in the list correspond to the homological dimensions, and the
+            integers of each row in each array correspond to: (index of positive vertex).
+        """
+
+        assert self.pcohptr != NULL, "compute_persistence() must be called before vertices_of_persistence_pairs()"
+
+        cdef vector[vector[int]] persistence_result
+        output = [[],[]]
+        with nogil:
+            persistence_result = self.pcohptr.vertices_of_cubical_persistence_pairs()
+        pr = np.array(persistence_result)
+
+        ess_ind = np.argwhere(pr[:,2] == -1)[:,0]
+        ess = pr[ess_ind]
+        max_h = max(ess[:,0])+1 if len(ess) > 0 else 0
+        for h in range(max_h):
+            hidxs = np.argwhere(ess[:,0] == h)[:,0]
+            output[1].append(ess[hidxs][:,1])
+
+        reg_ind = np.setdiff1d(np.array(range(len(pr))), ess_ind)
+        reg = pr[reg_ind]
+        max_h = max(reg[:,0])+1 if len(reg) > 0 else 0
+        for h in range(max_h):
+            hidxs = np.argwhere(reg[:,0] == h)[:,0]
+            output[0].append(reg[hidxs][:,1:])
+
+        return output
```

Comment:
Not sure about all the processing we are doing on the `persistence_result` to get the final returned `output`... I guess it's similar to `cofaces_of_persistence_pairs`...